worlds/spelunker/Rules.py

```diff
@@ -0,0 +1,116 @@
+from worlds.generic.Rules import set_rule
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from . import SpelunkerWorld
+
+
+def set_rules(world: "SpelunkerWorld") -> None:
+    player = world.player
+
+    set_rule(world.multiworld.get_location("B1F - Past Boulder", player), lambda state: state.has("Dynamite", player, 1))
+    set_rule(world.multiworld.get_location("B1F - Side Shaft Ledge", player), lambda state: state.has("Dynamite", player, 1))
+    set_rule(world.multiworld.get_location("B1F - Side Shaft Mound", player), lambda state: state.has("Dynamite", player, 1))
+    set_rule(world.multiworld.get_location("B1F - Side Shaft Bottom Left", player), lambda state: state.has("Dynamite", player, 1))
+    set_rule(world.multiworld.get_location("B1F - Side Shaft Bottom Right", player), lambda state: state.has("Dynamite", player, 1))
+    set_rule(world.multiworld.get_location("B1F - Side Shaft Ladder Right", player), lambda state: state.has("Dynamite", player, 1))
+
+    set_rule(world.multiworld.get_location("B3F - Bat Item", player), lambda state: state.has("Flare", player, 1))
+    set_rule(world.multiworld.get_location("B3F - Ladder Left", player), lambda state: state.has("Flare", player, 1))
+    set_rule(world.multiworld.get_location("B3F - Bombable Wall", player), lambda state: state.has("Dynamite", player, 1) and state.has("Flare", player, 1))
+
+    set_rule(world.multiworld.get_location("B5F - Near Ramp", player), lambda state: state.has("Dynamite", player, 1) and state.has("Blue Key", player, 1))
+    set_rule(world.multiworld.get_location("B5F - Ladder Left", player), lambda state: state.has("Dynamite", player, 1) and state.has("Blue Key", player, 1))
+    set_rule(world.multiworld.get_location("B5F - Ladder Right", player), lambda state: state.has("Dynamite", player, 1) and state.has("Blue Key", player, 1))
+
+    set_rule(world.multiworld.get_location("B8F - Rope 1", player), lambda state: state.has("Dynamite", player, 2))
+    set_rule(world.multiworld.get_location("B8F - Rope 2", player), lambda state: state.has("Dynamite", player, 2))
+    set_rule(world.multiworld.get_location("B8F - Boulder Pit 1", player), lambda state: state.has("Dynamite", player, 2))
+    set_rule(world.multiworld.get_location("B8F - Boulder Pit 2", player), lambda state: state.has("Dynamite", player, 2))
+
+    set_rule(world.multiworld.get_location("B11F - Boulder Pit", player), lambda state: state.has("Dynamite", player, 2))
+
+    set_rule(world.multiworld.get_location("B10F - Solo Rope", player), lambda state: state.has("Flare", player, 1))
+    set_rule(world.multiworld.get_location("B10F - Platform Item", player), lambda state: state.has("Flare", player, 1))
+    set_rule(world.multiworld.get_location("B11F - Bombable Wall", player), lambda state: state.has("Dynamite", player, 2) and state.has("Flare", player, 1))

```

Comment:
Not sure if intentional, but these aren't in floor order like most of the rest are.
This is also happening with floors 19 and 20, floors 23 and 25, and with floors 28-30.